builtin/logical/pki/cert_util.go

```diff
@@ -255,16 +608,261 @@ func createCertificate(creationInfo *certCreationBundle) (*certutil.ParsedCertBu
 	if creationInfo.Usage&codeSigningUsage != 0 {
 		certTemplate.ExtKeyUsage = append(certTemplate.ExtKeyUsage, x509.ExtKeyUsageCodeSigning)
 	}
+	if creationInfo.Usage&emailProtectionUsage != 0 {
+		certTemplate.ExtKeyUsage = append(certTemplate.ExtKeyUsage, x509.ExtKeyUsageEmailProtection)
+	}
+
+	certTemplate.IssuingCertificateURL = creationInfo.URLs.IssuingCertificates
+	certTemplate.CRLDistributionPoints = creationInfo.URLs.CRLDistributionPoints
+	certTemplate.OCSPServer = creationInfo.URLs.OCSPServers
+
+	var certBytes []byte
+	if creationInfo.SigningBundle != nil {
```

Comment:
I've moved this logic into the calling function. If the signing bundle is nil, it now simply returns.